### PR TITLE
Dispatch an event when content is loaded

### LINKED ISSUE #7: [RFC] add an event on content loaded success
Hi,
first thx for such component :)

I may ask you if it would be interesting for this component to have a

## Before

```js
  fetch (): void {
    fetch(this.urlValue)
      .then((response: Response) => response.text())
      .then((html: string) => {
        this.element.innerHTML = html

        if (this.loadScriptsValue) {
          this.loadScripts()
        }
      })
  }
```

## After

```js
  fetch (): void {
    fetch(this.urlValue)
      .then((response: Response) => response.text())
      .then((html: string) => {
        this.element.innerHTML = html

        this.element.dispatchEvent(
            new CustomEvent('content-loaded', { bubbles: true })
        );

        if (this.loadScriptsValue) {
          this.loadScripts()
        }
      })
  }
```

thank you,

### PR DESCRIPTION
**Pull request by 94noni**
resolve #7 

Some question while here, would it also be nice to handle, in the `fetch` a new `catch` and also dispatch?
For the event name, feel free to challenge :)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -75,6 +75,10 @@ export default class extends Controller {
       .then((html: string) => {
         this.element.innerHTML = html
 
+        this.element.dispatchEvent(
```
Comment (guillaumebriday-pa) [inline src/index.ts]: What about using the built in dispatch mechanism? https://stimulus.hotwired.dev/reference/controllers#cross-controller-coordination-with-events

Comment (94noni) [reply src/index.ts]: Yes i can swap to this

Comment (guillaumebriday) [reply src/index.ts]: Thanks!

Comment (guillaumebriday) [reply src/index.ts]: I think we can also add an event when fetch starts and when it fails as well

What do you think?

Comment (94noni) [reply src/index.ts]: Yes as asked in the PR desc, it could be great
Will see to add such event, using the stimulus dispatch

**Comment by 94noni** [comment]
@guillaumebriday-pa 👋🏻 hello, PR is ready again for review
cheers!

**Comment by 94noni** [comment]
friendly ping @guillaumebriday @guillaumebriday-pa

**Comment by 94noni** [comment]
friendly ping @guillaumebriday @guillaumebriday-pa

**Comment by guillaumebriday** [comment]
I changed the code a bit to stay consistent with rails-ujs conventions but here you go https://github.com/stimulus-components/stimulus-content-loader/releases/tag/v4.2.0

**Comment by 94noni** [comment]
@guillaumebriday fantastic ! thx

**Pull request merged**